haas/api.py

```diff
@@ -529,56 +565,111 @@ def network_delete(network):
     db = model.Session()
     network = _must_find(db, model.Network, network)
 
-    if network.nics:
+    if len(network.attachments) != 0:
         raise BlockedError("Network still connected to nodes")
     if network.hnics:
         raise BlockedError("Network still connected to headnodes")
-    if network.scheduled_nics:
-        raise BlockedError("Network scheduled to become connected to nodes.")
+    if len(network.scheduled_nics) != 0:
+        raise BlockedError("There are pending actions on this network")
     if network.allocated:
-        driver_name = cfg.get('general', 'driver')
-        driver = importlib.import_module('haas.drivers.' + driver_name)
-        driver.free_network_id(db, network.network_id)
+        get_network_allocator().free_network_id(db, network.network_id)
 
     db.delete(network)
     db.commit()
 
 
-                            # Port code #
-                            #############
+@rest_call('GET', '/network/<network>')
+def show_network(network):
+    """Show details of a network.
+
+    Returns a JSON object representing a network.
+    The object will have at least the following fields:
+        * "name", the name/label of the network (string).
+        * "creator", the name of the project which created the network, or
+          "admin", if it was created by an administrator.
+    It may also have the fields:
+        * "access" -- if this is present, it is the name of the project which
+          has access to the network. Otherwise, the network is public.
+    """
+    db = model.Session()
+    allocator = get_network_allocator()
 
+    network = _must_find(db, model.Network, network)
+    return json.dumps({
+        'name': network.label,
+        'channels': allocator.legal_channels_for(db, network.network_id),
+    })
```

Comment:
I feel like at some point there'd been a bit of debate as to what we actually wanted to list here, and came to this? I don't really recall. leaving it out means it's easier to change later. At some point we should take advantage of the fact that we can have option/structure arguments to make that whole array of options more intuitive.